app/screens/channel/channel.js

```diff
@@ -247,9 +249,10 @@ export default class Channel extends PureComponent {
 
         loadChannelsIfNecessary(teamId).then(() => {
             loadProfilesAndTeamMembersForDMSidebar(teamId);
-            selectInitialChannel(teamId);
-        }).catch(() => {
-            selectInitialChannel(teamId);
+            selectInitialChannel(teamId, app.startAppFromPushNotification);
```

Comment:
the reason that we need to know this is because the selectInitialChannel will select the channel for the team and mark the previous one (same one) as read, causing the `new message indicator` to disappear. Actually I think we need to change this as if it was open from a push notification to not select the initial channel at all

Reply:
The way I used to picture it was that we'd have a separate Entry screen that comes up before the Channel screen, and that Entry screen would handle setting the current channel/team ID based on whether you came from a push notification or based on your last viewed channel/team before the app was last closed. I'm not sure if you had that specifically in mind, but I do like the idea of setting the channel ID from the notification before we get to this point.